Buildings/Fluid/HeatExchangers/BaseClasses/WetCoilUARated.mo

```diff
@@ -0,0 +1,192 @@
+within Buildings.Fluid.HeatExchangers.BaseClasses;
+model WetCoilUARated
+  "Model that calculates the UA-value from cooling coil data at rated conditions."
+
+  replaceable package MediumA=Buildings.Media.Air
+    constrainedby Modelica.Media.Interfaces.PartialMedium
+    "Air-side medium";
+  replaceable package MediumW=Buildings.Media.Water
+    constrainedby Modelica.Media.Interfaces.PartialMedium
+    "Water-side medium";
+
+  parameter Boolean use_UA_nominal = false
+    "Set to true to specify UA_nominal, or to false to use nominal conditions"
+    annotation (Evaluate=true,
+                Dialog(group="Nominal thermal performance"));
+  parameter Modelica.SIunits.Temperature TAirIn
+    "Air inlet temperature at a rated condition";
+  parameter Modelica.SIunits.Temperature TAirOut
+    "Air outlet temperature  at a rated condition";
+  parameter Modelica.SIunits.MassFraction X_wAirIn
+    "Mass fraction of water in inlet air at a rated condition";
+  parameter Modelica.SIunits.Temperature TWatIn
+    "Water inlet temperature at a rated condition";
+  parameter Modelica.SIunits.Temperature TWatOut
+    "Water outlet temperature at a rated condition";
+  parameter Modelica.SIunits.MassFlowRate mAir_flow
+    "Air mass flow rate at a rated condition";
+  parameter Modelica.SIunits.MassFlowRate mWat_flow
+    "Water mass flow rate at a rated condition";
+  parameter Modelica.SIunits.ThermalConductance UA
+    "UA is for the overall coil (i.e., both sides)";
+  parameter Real r_nominal(min=0, max=1)
+    "Ratio between air-side and water-side convective heat transfer at nominal condition";
+protected
+  constant Modelica.SIunits.SpecificEnthalpy hfg=
```

Comment:
All parameters, constants and variables must have a documentation string.

Reply:
Revised